code/modules/power/supermatter/supermatter_cascade_components.dm

```diff
@@ -117,16 +135,18 @@
 /obj/cascade_portal/proc/consume(atom/movable/consumed_object)
 	if(isliving(consumed_object))
 		var/list/arrival_turfs = get_area_turfs(/area/centcom/central_command_areas/evacuation)
-		var/turf/arrival_turf = pick(arrival_turfs)
+		var/turf/arrival_turf
+		do
+			arrival_turf = pick(arrival_turfs)
+		while(!is_safe_turf(arrival_turf))
```

Comment:
please make sure this doesn't crash the server by using something like pick_n_take

Reply:
pick_n_take is exactly what I was looking for but could not find here, good stuff